### PR TITLE
Update Rigid Body Actions and Input File

### PR DESCRIPTION
**Pull request by itgreenquist**
Updates rigidbodymotion objects to:
- Include more options/capabilities in MatVecRealGradAuxKernelAction
- Reduce number of times "etas" has to be entered into input files
- Improve convergence for AC_CH_Multigrain.i
#6725

### CODE REVIEW

File: libmesh [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit a04c0c04d13ef83849a74de9f0ad24a8342d3764
+Subproject commit 676407ba6c856096e0c74303697d06a837ac9d55
```
Comment (dschwen) [inline libmesh]: Please remove the libmesh submodule commit

File: libmesh [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit a04c0c04d13ef83849a74de9f0ad24a8342d3764
+Subproject commit e90f9999dca958106ceb46c0052e2e2d95359d46
```
Comment (itgreenquist) [inline libmesh]: @dschwen How do I get rid of this?

**Comment by dschwen** [comment]
Do

```
git checkout HEAD~1 -- libmesh
git submodule update
git commit -a --amend
git push -f
```

that should work.

**Comment by dschwen** [comment]
F you @moosebuild

**Comment by itgreenquist** [comment]
Thanks @dschwen

File: modules/phase_field/src/action/MatVecRealGradAuxKernelAction.C [hunk 3]
```diff
@@ -31,38 +34,55 @@ MatVecRealGradAuxKernelAction::MatVecRealGradAuxKernelAction(const InputParamete
 void
 MatVecRealGradAuxKernelAction::act()
 {
-  std::vector<unsigned int> _op_num = getParam<std::vector<unsigned int> >("op_num");
+  unsigned int _op_num = getParam<unsigned int>("op_num");
   std::vector<std::string> _var_name_base = getParam<std::vector<std::string> >("var_name_base");
```
Comment (dschwen) [inline modules/phase_field/src/action/MatVecRealGradAuxKernelAction.C]: This shadows the member, which is **not** a vector type. Please remove the member and remove the leading `_` here.

File: modules/phase_field/src/kernels/GrainRigidBodyMotionBase.C [hunk 4]
```diff
@@ -12,7 +12,7 @@ InputParameters validParams<GrainRigidBodyMotionBase>()
   InputParameters params = validParams<Kernel>();
   params.addClassDescription("Base class for adding rigid mody motion to grains");
   params.addRequiredCoupledVar("c", "Concentration");
-  params.addRequiredCoupledVar("v", "Array of coupled variable names");
+  params.addRequiredCoupledVarWithAutoBuild("v", "var_name_base", "op_num", "Array of coupled variable names");
```
Comment (dschwen) [inline modules/phase_field/src/kernels/GrainRigidBodyMotionBase.C]: Does this work with `var_name_base` being a vector of strings?! I'd think it only takes the first supplied string. I'll check.

**Comment by andrsd** [comment]
One general comment regarding code standard:
> Member variables start with `_`, local variables do not.

It seems some local variables start with `_` here, I recommend to fix that as well...

File: modules/phase_field/include/action/MatVecRealGradAuxKernelAction.h [hunk 5]
```diff
@@ -17,7 +17,10 @@ class MatVecRealGradAuxKernelAction: public Action
   virtual void act();
 
 private:
+  AuxVariableName _div_var;
   std::string _var_name_base;
```
Comment (dschwen) [inline modules/phase_field/include/action/MatVecRealGradAuxKernelAction.h]: This is still there (and unused)

File: modules/phase_field/src/action/MatVecRealGradAuxKernelAction.C [hunk 6]
```diff
@@ -31,38 +34,55 @@ MatVecRealGradAuxKernelAction::MatVecRealGradAuxKernelAction(const InputParamete
 void
 MatVecRealGradAuxKernelAction::act()
 {
-  std::vector<unsigned int> _op_num = getParam<std::vector<unsigned int> >("op_num");
   std::vector<std::string> _var_name_base = getParam<std::vector<std::string> >("var_name_base");
```
Comment (dschwen) [inline modules/phase_field/src/action/MatVecRealGradAuxKernelAction.C]: This local variable still starts with `_`

File: modules/phase_field/src/action/MultiAuxVariablesAction.C [hunk 7]
```diff
@@ -32,31 +33,35 @@ MultiAuxVariablesAction::MultiAuxVariablesAction(InputParameters params) :
...
+*/
+
 void
 MultiAuxVariablesAction::act()
 {
   MooseEnum order = getParam<MooseEnum>("order");
   MooseEnum family = getParam<MooseEnum>("family");
-  std::vector<unsigned int> op_num = getParam<std::vector<unsigned int> >("op_num");
+  unsigned int op_num = getParam<unsigned int>("op_num");
   std::vector<std::string> var_name_base = getParam<std::vector<std::string> >("var_name_base");
```
Comment (dschwen) [inline modules/phase_field/src/action/MultiAuxVariablesAction.C]: Is there a member in `MultiAuxVariablesAction.h` that could be removed now? Please compile in `dbg` mode and check if you have _any_ compile warnings in those files (and fix them).

File: modules/phase_field/src/action/MultiAuxVariablesAction.C [hunk 8]
```diff
@@ -32,31 +33,35 @@ MultiAuxVariablesAction::MultiAuxVariablesAction(InputParameters params) :
...
   std::set<SubdomainID> blocks = getSubdomainIDs();
+  unsigned int _dim = getParam<unsigned int>("dim");
```
Comment (dschwen) [inline modules/phase_field/src/action/MultiAuxVariablesAction.C]: Another local variable that starts with `_`

File: modules/phase_field/src/kernels/GrainRigidBodyMotionBase.C (hunk 4 above)
Comment (itgreenquist) [inline modules/phase_field/src/kernels/GrainRigidBodyMotionBase.C]: I think I've taken out all of the instances of `var_name_base` being a vector of strings. It turned out it did not work very well with the type of auxkernels being used here.

**Comment by dschwen** [comment]
I have invalidated the failed test. It was an unrelated timeout.

File: modules/phase_field/include/action/MultiAuxVariablesAction.h [hunk 9]
```diff
@@ -20,8 +20,6 @@ class MultiAuxVariablesAction: public AddAuxVariableAction
 
   virtual void act();
```
Comment (dschwen) [inline modules/phase_field/include/action/MultiAuxVariablesAction.h]: empty line

File: modules/phase_field/src/action/MultiAuxVariablesAction.C (hunk 7 above)
Comment (itgreenquist) [inline modules/phase_field/src/action/MultiAuxVariablesAction.C]: @jwpeterson took a few unused members out recently. So the only warnings I am seeing when compiling in `dbg` mode are in the tensor mechanics module.

File: modules/phase_field/include/action/MatVecRealGradAuxKernelAction.h [hunk 10]
```diff
@@ -17,7 +17,9 @@ class MatVecRealGradAuxKernelAction: public Action
   virtual void act();
 
 private:
```
Comment (dschwen) [inline modules/phase_field/include/action/MatVecRealGradAuxKernelAction.h]: this `private:` is not necessary, please make it `protected:` instead.

**Comment by dschwen** [comment]
:+1: I've invalidated the app tests. Two Bison plasticity tests were exodiffing. It seems for reasons unrelated to this PR. I'll merge when it passes.

**Pull request merged**